docs/tutorials/using_apis/connectivity/tls_tutorial.md

```diff
@@ -0,0 +1,173 @@
+<h1 id="tls-tutorial">TLS Tutorial</h1>
+
+Since Mbed OS 5.11, the [IP networking interface](../reference/ip-networking.html) has been extended to include [TLSSockets](../apis/tlssocket.html), which behave similarly to normal TCP sockets but automatically use Mbed TLS to set up a TLS connection to the server. This tutorial, based on [our blog entry](https://os.mbed.com/blog/entry/Adding-TLS-Sockets-to-Mbed-OS/), will help you understand and use TLS encryption in Mbed OS.
+
+In particular, this tutorial:
+
+- Introduces the basic concepts of CA certificates.
+- Explains how to practically obtain them and embed them in your application using TLSSockets.
+- Points to further reading on client's key and certificate verification.
+
+## Certificate Authorities
+
+Unlike desktop operating systems, such as Windows or macOS, Mbed OS does not have a central list of trusted Certificate Authorities (CAs). This means that when you want to talk to a server through TLS, you need to provide the list of root CAs, yourself. You can find the root CAs quickly through your browser or through OpenSSL.
+
+### Browser
+
+You can use any browser to find the root CAs, but this tutorial uses Firefox:
+
+<span class="images">![](https://s3-us-west-2.amazonaws.com/mbed-os-docs-images/root-ca-selection.png "Getting root CA in Firefox")<span>Getting root CA in Firefox.</span></span>
+
+1. Use Firefox to go to a page that uses HTTPS and is hosted on the same domain as the server you want to talk to over a TLS Socket.
+1. Click `Tools` > `Page Info`.
+1. Click `Security`.
+1. Click `View Certificate`.
+1. Choose the `Details` tab.
+1. Click on the top item in the certificate hierarchy; this is the root CA.
+1. Click `Export`.
+1. This gives you a `.crt` file. Store it inside your project, so you can find it later.
+
+### OpenSSL
+
+1. Open a terminal and run:
+
+   ```
+   $ openssl s_client -connect os.mbed.com:443 -showcerts
+   ```
+
+   (Replace `os.mbed.com:443` with your host and port.)
+
+1. Look for the last occurance of `-----BEGIN CERTIFICATE-----`.
+1. Copy everything from `-----BEGIN CERTIFICATE-----` to `-----END CERTIFICATE-----`, and store it in a file. 
+
+<span class="images">![](https://s3-us-west-2.amazonaws.com/mbed-os-docs-images/tlssocket02.png "Getting root CA via OpenSSL")<span>Getting root CA through OpenSSL.</span></span>
+
+## Placing the CA in code
+
+To use the CA in code, you need to store it in a string and escape the content. For example:
```

Comment:
Is "escape" the right word here? What does it mean in this context?

Reply:
You are right, I improved the wording to match exactly what the example is showing.